### PR TITLE
Document logical file statistics for Delta Lake

### PR DESCRIPTION
**Pull request by findinpath**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by mosabua — COMMENTED** [review]
Can you confirm the modification from a technical point of view @alexjo2144 please.. if that side is good .. I can merge

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -726,10 +726,11 @@ following sections:
...
 ^^^^^^^^^^^^^^^^
 
-Use :doc:`/sql/analyze` statements in Trino to populate the table statistics in
-Delta Lake. Data size and number of distinct values (NDV) statistics are
-supported; whereas minimum value, maximum value, and null value count statistics
-are not supported. The :doc:`cost-based optimizer
+Use :doc:`/sql/analyze` statements in Trino to populate data size and
+number of distinct values (NDV) extended table statistics in Delta Lake.
+The minimum value, maximum value, value count and null value count
+statistics are computed on the fly out of the *logical files* statistics
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.rst:732]: I think users can't understand the meaning of "logical files". Can we rephrase that?

Comment (findinpath) [reply docs/src/main/sphinx/connector/delta-lake.rst]: Changed to _transaction log_ - it is a more generic notion and doesn't go into implementation details.

**Review by mosabua — APPROVED** [review]
Looks good now. Thanks @findinpath

**Pull request merged**